src/containers/DiskSpace/DiskSpaceContainer.tsx

```diff
@@ -1,49 +1,79 @@
-import React, {ReactElement, useState} from "react";
+import React, {ReactElement, useState, useEffect} from "react";
 import {ChartHeader} from "../../components/ChartHeader/ChartHeader";
 import {DiskSpaceChart} from "./DiskSpaceChart";
+import {useSelector, useDispatch} from "react-redux";
+import {RootState} from "../../app/rootReducer";
+import {getDiskInfo} from "../Dashboard/NodeSlice";
+import {INodeDiskState} from "../Dashboard/NodeInterface";
 
-export type DiskSpaceProps = {
+export interface IDiskSpaceProps {
+    selectedNodeIndex: number;
+}
+
+export type DiskSpaceDataProps = {
     date: string;
     free: number;
     taken: number;
 };
 
-export const DiskSpace = (): ReactElement => {
-    const data: DiskSpaceProps[] = [
-        {date: "2020-01-23T14:43:42.856Z", free: 81, taken: 22},
-        {date: "2020-01-24T14:43:42.856Z", free: 102, taken: 68},
-        {date: "2020-01-25T14:43:42.856Z", free: 1343, taken: 1001},
-        {date: "2020-01-26T14:43:42.856Z", free: 200, taken: 20},
-        {date: "2020-01-27T14:43:42.856Z", free: 1234, taken: 1092},
-        {date: "2020-01-28T14:43:42.856Z", free: 926, taken: 925},
-    ];
+export const DiskSpace: React.FC<IDiskSpaceProps> = ({selectedNodeIndex}: IDiskSpaceProps): ReactElement => {
+    const formatDiskData = (diskData: Array<INodeDiskState>): Array<DiskSpaceDataProps> => {
+        const formatedData: Array<DiskSpaceDataProps> = [];
+        diskData.forEach(e => {
+            formatedData.push({
+                date: e.updatedAt,
+                free: Math.round(e.freeSpace / 1000000000),
```

Comment:
This is not good math and can be very inprecise on big numbers as freeSpace is also a string. What are you trying to do here?